src/Syntax.elm

```diff
@@ -38,8 +32,8 @@ type SyntaxType
     | BooleanLiteral
     | Blank
     | Var
-    | Reference Hash
-    | Referent Hash
+    | TypeReference Hash
+    | TermReference Hash
```

Comment:
@runarorama as far as I could tell the changes were `Reference` -> `TypeReference`, `Referent` to `TermReference` and `HashQualifier HashQualified` -> `HashQualifier String`—did I miss any other?

Reply:
That should cover it!